driver/src/price_estimation/dexag/mod.rs

```diff
@@ -1 +1,178 @@
 mod api;
+
+use super::{PriceSource, Token};
+use crate::models::TokenId;
+use anyhow::{anyhow, Result};
+use api::{DexagApi, DexagHttpApi};
+use std::collections::HashMap;
+
+pub struct DexagClient<Api> {
+    api: Api,
+    // Maps Token::symbol to Token.
+    // This is cached in the struct because we don't expect it to change often.
+    tokens: HashMap<String, api::Token>,
+    stable_coin: api::Token,
+}
+
+impl DexagClient<DexagHttpApi> {
+    /// Create a DexagClient using DexagHttpApi as the api implementation.
+    #[allow(dead_code)]
+    pub fn new() -> Result<Self> {
+        let api = DexagHttpApi::new()?;
+        Self::with_api(api)
+    }
+}
+
+impl<Api> DexagClient<Api>
+where
+    Api: DexagApi,
+{
+    pub fn with_api(api: Api) -> Result<Self> {
+        // We need to return prices in OWL but Dexag does not track it. OWL tracks
+        // USD so we use another stable coin as an approximate USD price.
+        const STABLE_COIN: &str = "DAI";
+
+        let tokens = api.get_token_list()?;
+        let mut tokens: HashMap<String, api::Token> = tokens
+            .into_iter()
+            .map(|token| (token.symbol.clone(), token))
```

Comment:
nit: feels like if we implemented hash and equality on token to just look at the symbol we wouldn't need a map here, but could use a set instead.

Reply:
We need to store `tokens` in the struct anyway because it is used to look up `Api::Token` by `Token::symbol`.